### ISSUE TITLE
add ext save

### ISSUE THREAD

**Issue by smallevil** [rhsimplex/image-match#25]
I modified the code, adds a field.
You can add pictures at the same time adding a JSON text description.
At the same time of reading can be obtained.
But I'm not familiar with git, I do not know how to submit to you.
So, playing a compressed package. You see if you can merge into it.
I hope that will not increase your work.

download url:
http://7jpsbs.com1.z0.glb.clouddn.com/image-match.tar.gz

**Comment by rhsimplex** [comment]
Thanks for the effort @smallevil I'll have a look =)

**Comment by rhsimplex** [comment]
I created a pull request with your code: #26 looks good, but I have to finish the integration tests #19 before merging anything.  Thanks again for your help!

**Comment by smallevil** [comment]
You look forward to merge into it.
This project is like

**Comment by pablodgonzalez** [comment]
Hi, @rhsimplex and @smallevil .
Today, I started to use this project, in few minutes, I created un docker-compose and loaded data and tested it.
It was awesome, but at the last hours of day I understand that we need searching with metadata too.
I'd like search by image and metadata, because I use different enough image and I can filter about 90% data, but still many results which to get most asserted results.
I can implement that but the best thing is wait for the integration of pull request.

**Comment by smallevil** [comment]
@pablodgonzalez 
I have a question:
If you are using metadata,
For example: a picture, it appears in the A site too, are B site appeared, in fact, I think both are added to the list can be found out.
If you are using metadata, will not cause the same picture can only be added to the library again?

**Comment by smallevil** [comment]
@pablodgonzalez 
Also:
If you do, the search speed will not be affected?

**Comment by rhsimplex** [comment]
Hi @pablodgonzalez! I'm back from vacation and can work on this now.

@smallevil there is currently no code to block duplicates.  You could make some metadata field the index (primary key) and protect against duplicates that way, since the index is unique.  This shouldn't slow things down.  Or you could reject adding very similar images if you like.

**Comment by smallevil** [comment]
@rhsimplex 
Thank you, so just fine.
We look forward to publish as soon as possible.
I've been waiting for after the release, I can use.
I'm impatient

**Comment by rhsimplex** [comment]
@pablodgonzalez @smallevil can you tell me how you envision the metadata search to work? We have added the metadata field (thanks @smallevil), but should searching the metadata search be separate? Should it be mixed into the `dist` score? Should a schema be enforced or is anything permissible as metadata?

In my opinion, anything should be permissible as metadata, but it is up to the user to implement their own queries.

**Comment by smallevil** [comment]
Can you tell me the next version probably Published Why?
Because my side of the project is also waiting for you in the new version can store JSON. It has been stalled for some time.

**Comment by rhsimplex** [comment]
I can't merge this PR into master because the metadata fields should be separated from the image search fields. I still need to modify your code a bit. No ETA unfortunately.

In the meantime, you can use your code by doing:

```
git clone https://github.com/ascribe/image-match.git
cd image-match/
git checkout ba881a8b7f4f33b7a1d87d6b20724e58639e097e
```

So you don't need to wait on me
